File: Mega_Contest_1/Editorials/Problem-3_Maximum-Sum/solution2.py
def maxSum(a, b, k, n): 
    a.sort() 
    b.sort() 
    i = 0
    j = n - 1
    
    while i < k: 
        if (a[i] < b[j]): 
            a[i], b[j] = b[j], a[i]             
        else: 
            break
        i += 1
        j -= 1
    sum = 0
    for i in range (n): 
        sum += a[i]      
    return(sum) 


tc = int(input())
for i in range(tc):
    kn = input()
    l = list(kn.split(' '))
    l = list(map(int,l))
    #print(tc,kn,l)
    k = l[1]
    n = l[0]
    a1 = input() 
    b1 = input()
    a = list(a1.split(' '))
    b = list(b1.split(' '))
    a = list(map(int,a))
    b = list(map(int,b)) 

    print(maxSum(a, b, k, n)) 

File: Mega_Contest_1/Editorials/Problem-3_Maximum-Sum/solution2.cpp
#include <cmath>
#include <cstdio>
#include <vector>
#include <iostream>
#include <algorithm>
using namespace std;


int main() {
    /* Enter your code here. Read input from STDIN. Print output to STDOUT */   
    int t;
    cin >> t;
    while(t--) {
        int n, k;
        cin >> n >> k;
        int a[n], b[n];
        for(int i=0;i<n;i++) 
            cin >> a[i];
        for(int i=0;i<n;i++) 
            cin >> b[i];
        sort(a, a+n);
        sort(b, b+n);
        int i = 0;
        int sum = 0;       
        if(b[n-1] <= a[0]) {
            for(i=0;i<n;i++) {
                sum += a[i];
            }
        }
        else {
            for(int i=0;i<k;i++) {
                if(b[n-i-1] > a[i]) {
                    a[i] = b[n-i-1];
                }
                else {
                    break;
                }
            }
            for(int i=0;i<n;i++) {
                sum += a[i];
            }
        }
        cout << sum << endl;
    }
    return 0;
}
